src/app/pages/home/components/cinema-details/cinema-details.component.ts

```diff
@@ -11,16 +15,24 @@ import { ActivatedRoute, Router } from '@angular/router';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class CinemaDetailsComponent implements OnInit, OnDestroy {
-  cinema?: ICinema;
+  cinema: ICinema;
+
+  sessions: ISession[];
+
+  groupedSessionsByDate: IGroupedSessions[];
 
   collapsing = true;
 
   cinemaSub: Subscription;
 
+  sessionSub: Subscription;
+
```

Comment:
When you have more than 1 subscription it would be better to use another method for unsubscribing
Have a look at takeUntil operator

Reply:
Btw, you forgot to unsubscribe 😁